NCS.DSS.Goal/GetGoalByIdHttpTrigger/Function/GetGoalByIdHttpTrigger.cs

```diff
@@ -44,109 +39,105 @@ public GetGoalByIdHttpTrigger(IResourceHelper _resourceHelper, IHttpRequestHelpe
         [Display(Name = "Get", Description = "Ability to retrieve an individual Goals for the given customer")]
         public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "Customers/{customerId}/Interactions/{interactionId}/ActionPlans/{actionPlanId}/Goals/{goalId}")] HttpRequest req, string customerId, string interactionId, string actionPlanId, string goalId)
         {
+            _logger.LogInformation("Function {FunctionName} has been invoked", nameof(GetGoalByIdHttpTrigger));
 
             var correlationId = httpRequestHelper.GetDssCorrelationId(req);
             if (string.IsNullOrEmpty(correlationId))
             {
-                log.LogInformation("Unable to locate 'DssCorrelationId' in request header");
+                _logger.LogInformation("Unable to locate 'DssCorrelationId' in request header");
             }
 
             if (!Guid.TryParse(correlationId, out var correlationGuid))
             {
-                log.LogInformation("Unable to parse 'DssCorrelationId' to a Guid");
+                _logger.LogInformation("Unable to parse 'DssCorrelationId' to a Guid. CorrelationId: {CorrelationId}", correlationId);
                 correlationGuid = Guid.NewGuid();
             }
 
-            log.LogInformation($"DssCorrelationId: [{correlationGuid}]");
-
-
             var touchpointId = httpRequestHelper.GetDssTouchpointId(req);
             if (string.IsNullOrEmpty(touchpointId))
             {
-                var response = new BadRequestObjectResult(HttpStatusCode.BadRequest);
-                log.LogWarning($"Response Status Code: [{response.StatusCode}]. Unable to locate 'TouchpointId' in request header");
-                return response;
+                _logger.LogInformation("Unable to locate 'TouchpointId' in request header.");
+                return new BadRequestObjectResult(HttpStatusCode.BadRequest);
             }
 
-            log.LogInformation($"Get Action Plan By Id C# HTTP trigger function  processed a request. By Touchpoint: [{touchpointId}]");
+            _logger.LogInformation("Header validation has succeeded. Touchpoint ID: {TouchpointId}. Correlation GUID: {CorrelationGuid}", touchpointId, correlationGuid);
```

Comment:
I would either move this down to line 88 and change to "input validation" instead of header or i would add another log on 88 to state body and url validation is complete